src/Repository/SearchInputRepository.php

```diff
@@ -0,0 +1,358 @@
+<?php
+/**
+ * Clarifai SearchInput
+ *
+ * @category Input
+ * @package  Clarifai
+ * @author   Darryn Ten <darrynten@github.com>
+ * @license  MIT <https://github.com/darrynten/clarifai-php/LICENSE>
+ * @link     https://github.com/darrynten/clarifai-php
+ */
+
+namespace DarrynTen\Clarifai\Repository;
+
+use DarrynTen\Clarifai\Entity\Concept;
+use DarrynTen\Clarifai\Entity\Input;
+
+/**
+ * Single Clarifai Input
+ *
+ * @package Clarifai
+ */
+class SearchInputRepository extends InputRepository
+{
+    /**
+     * parameter for input Concepts search
+     */
+    const INPUT_CONCEPTS = 'input_concepts';
+
+    /**
+     * parameter for output Concepts search
+     */
+    const OUTPUT_CONCEPTS = 'output_concepts';
+
+    /**
+     * parameter for input Metadata search
+     */
+    const METADATA = 'metadata';
+
+    /**
+     * parameter for input image search
+     */
+    const IMAGES = 'images';
+
+    /**
+     * parameter for reverse image search
+     */
+    const REVERSED_IMAGES = 'reversed_images';
+
+    /**
+     * Searches Inputs
+     *
+     * @param array $params
+     *
+     * @return Input[] array
+     */
+    public function search($params)
+    {
+        $searchResult = $this->getRequest()->request(
+            'POST',
+            'searches',
+            $this->getSearchQuery($params)
+        );
+
+        return $this->getInputsFromSearchResult($searchResult);
+    }
+
+    /**
+     * @param array $params
+     *
+     * @return array
+     */
+    public function getSearchQuery($params)
+    {
+        $data['query'] = [];
+        $data['query']['ands'] = [];
+
+        if (isset($params[self::INPUT_CONCEPTS])) {
```

Comment:
array_key_exists will return true if $params[self::INPUT_CONCEPTS] === null and as a result we'll have one empty $data['query']['ands'] in our query and may cause error. In fact we do not need this key empty - only with data inside.